### PR TITLE
Extend bone override capabilities

### PR DESCRIPTION
**Pull request by appgurueu**
#### Description

Implements **bone overrides** supporting **position, rotation and *scale***, all of which can be **relative and interpolated**.

#### To do

- [ ] Use inline functions/loops instead of copypasted code
- [ ] Some renames from bone position -> bone override
- [ ] Deprecate `set_bone_position` and `get_bone_position`

#### How to test

Use the new documented API.

### CODE REVIEW

**Comment by theviper121** [comment]
I'm just curious. What is the use case for scaling a bone?

**Comment by hecktest** [comment]
> What is the use case for scaling a bone?

Boob slider =]

Among other things, it's good for FX. And just because, hell, do you need a reason? It's called freedom. You can scale whole objects so why not bones?

@appgurueu 
Is it possible to override one of the transform channels for a bone while still animating the others? That would be useful.

**Comment by appgurueu** [comment]
> @appgurueu
> Is it possible to override one of the transform channels for a bone while still animating the others? That would be useful.

A value of `[property] = { x = 0, y = 0, z = 0, absolute = false, interpolation = 0 }` will do the job.

**Comment by hecktest** [comment]
That's kinda hacky. Why not property=false/nil?

Also, there are more possible spaces than absolute/relative (to what?), maybe that should be a string param on the lua side and an enum on the engine side, to allow further extension.

Check this out:
https://github.com/minetest/minetest/issues/10301#issuecomment-677520051

**Comment by appgurueu** [comment]
> That's kinda hacky. Why not property=false/nil?

Properties default to what I have mentioned above. If you pass `{position = ...}`, rotation and scale will remain unchanged.

**Comment by hecktest** [comment]
Oh, so nil is already handled like that. Well, I like this.
Maybe the word "position" in the code should be changed to "transform" now that the full transform is being handled.

**Comment by hecktest** [comment]
How does this interact with #10705? Does setting an all-nil override make @theviper121 's function redundant?

**Comment by appgurueu** [comment]
> How does this interact with #10705? Does setting an all-nil override make @theviper121 's function redundant?

Yes, although it doesn't delete the override if doing so (could implement tho). Instead it justs sets all (0,0,0)-relative overrides (which equals no override).

**Comment by hecktest** [comment]
Well uh, I'm thoroughly confused but let me just describe the best possible API:
- `nil `disables the override on a given channel
- `0,0,0` just freezes the property at that position, and isn't treated as a magic value

Can you do that?

**Comment by appgurueu** [comment]
Latest commit makes #10705 obsolete

**Comment by rubenwardy** [comment]
Closing as WIP with no progress

File: doc/lua_api.txt [hunk 1]
```diff
@@ -6215,9 +6215,17 @@ object you are working with still exists.
...
     * `bone`: string
-    * `position`: `{x=num, y=num, z=num}` (relative)
+    * `position`: `{x=num, y=num, z=num}`
     * `rotation`: `{x=num, y=num, z=num}`
 * `get_bone_position(bone)`: returns position and rotation of the bone
+<!-- TODO compat -->
+* `set_bone_override(bone, override)`
+    * `bone`: string
+    * `override`: `{ position = property, rotation = property, scale = property }`
+      * `property`: `{ x = 0, y = 0, z = 0, interpolation = 0 --[[in seconds]], absolute = false}`
```
Comment (appgurueu) [inline doc/lua_api.txt:6225]: this should be renamed to interpolation_duration

**Comment by appgurueu** [comment]
Revival of this is at https://github.com/minetest/minetest/pull/12388

**Pull request closed without merging**